httpbp/client_middlewares.go

```diff
@@ -370,21 +371,58 @@ func PrometheusClientMetrics(serverSlug string) ClientMiddleware {
 // ClientBaseplateHeadersMiddleware is a middleware that forwards baseplate headers from the context to the outgoing request.
 //
 // If it detects any new baseplate headers set on the request, it will reject the request and return an error.
-func ClientBaseplateHeadersMiddleware(client string) ClientMiddleware {
+func ClientBaseplateHeadersMiddleware(client string, store SecretsStore, path string) ClientMiddleware {
+	getSigningSecret := func() *secrets.VersionedSecret {
+		secret, err := store.GetVersionedSecret(path)
+		if err != nil {
+			slog.Error(
+				"Failed to get secret:",
```

Comment:
also nit: for structured logging the `:` is less useful 😅
```suggestion
				"Failed to get secret",
```